openedx/xblock-lti-consumer
# fix: CCX LTI configuration compatibility

## Description

This PR adds compatibility fix for CCXs and external reusable configurations to the LTI consumer XBlock. This also includes changes to previously added fixes for CCXs using the LTI consumer XBlock. This changes allow the LTI consumer XBlock to sync values from the main CCX LTI configuration model to all of it's children CCX LTI configurations and vice versa.

## Type of Change

- [x] Remove previous CCX fixes.
- [x] Add model_to_dict function to utils module.
- [x] Add LtiConfiguration sync_configurations method.
- [x] Modify LtiConfiguration save method.
- [x] Include tests for all added and modified code.

## Testing

1. Create a course and add a LTI consumer XBlock.
2. Create a CCX of the previously created course. 
3. Launch the CCX course unit with the LTI consumer XBlock.
4. Go to http://localhost:18000/admin/lti_consumer/lticonfiguration/.
5. Both the main LTI configuration and it's children CCX configuration should have the same values.
6. Update any value on the main LTI configuration.
7. Go to the children CCX LTI configuration.
8. The values from the main LTI configuration should be reflected.
9. Delete both LTI configurations.
10. Launch the CCX course unit with the LTI consumer XBlock.
11. Launch the course unit with the main LTI consumer XBlock.
12. Go back to the LTI configuration admin and search for both newly created configurations.
13. Both LTI configurations values should be synced.
